src/librustc/middle/ty.rs

```diff
@@ -6112,7 +6086,7 @@ impl<'tcx> ctxt<'tcx> {
     }
 
     // Returns a list of `ClosureUpvar`s for each upvar.
-    pub fn closure_upvars(typer: &Typer<'tcx>,
+    pub fn closure_upvars<'a>(typer: &infer::InferCtxt<'a, 'tcx>,
```

Comment:
@eddyb yeah I was going to come back in a second pass a move a lot of the code on `InferCtxt` into methods, but I can do it in this patch as well.